src/player.cpp

```diff
@@ -579,7 +577,7 @@ void player::apply_persistent_morale()
 
     // Masochists get a morale bonus from pain.
     if (has_trait("MASOCHIST") || has_trait("MASOCHIST_MED") ||  has_trait("CENOBITE")) {
-        int bonus = pain / 2.5;
+        int bonus = get_pain() / 2.5;
```

Comment:
I think here should be `get_perceived_pain()` since it's impossible for masochists to enjoy the pain they cannot feel.